src/components/ChatContext.tsx

```diff
@@ -0,0 +1,98 @@
+import { Argument } from 'classnames'
+import { Channel } from 'phoenix'
+import React, { useEffect } from 'react'
+import { createContext, useContext, useState } from 'react'
+import { ChatHandler } from '.'
+import InputMethodTemplate from './elements/InputMethodTemplate'
+import Message, { Payload } from './elements/types'
+import ElementFactory from './elements'
+
+export function useChatProps() {
+  return useContext(ChatPropsContext)
+}
+
+export function useChatPropsUpdate() {
+  return useContext(ChatUpdateContext)
+}
+
+interface ChatProps {
+  channel: Channel & { hasMoreHistory: () => boolean; getMoreHistory: () => any }
+  conversationMeta: any
+  events: any[]
```

Comment:
Maybe at this point the events can be of type `events: Message<unknown>`, instead of any[]?